lib/classes/Service.js

```diff
@@ -33,23 +33,37 @@ class Service {
     const options = rawOptions || {};
     options.stage = options.stage || options.s;
     options.region = options.region || options.r;
-    const servicePath = that.serverless.config.servicePath;
+    const servicePath = this.serverless.config.servicePath;
 
     // skip if the service path is not found
     // because the user might be creating a new service
     if (!servicePath) {
       return BbPromise.resolve();
     }
 
-    let serverlessYmlPath = path.join(servicePath, 'serverless.yml');
-    // change to serverless.yaml if the file could not be found
-    if (!this.serverless.utils.fileExistsSync(serverlessYmlPath)) {
-      serverlessYmlPath = path
-        .join(this.serverless.config.servicePath, 'serverless.yaml');
-    }
+    // List of supported service filename variants.
+    // The order defines the precedence.
+    const serviceFilenames = [
```

Comment:
Yes, you're right. I will add them - also a precedence check would be beneficial.